### ISSUE TITLE
Rename destroy() to dispose()

### ISSUE THREAD

**Issue by reillyeon** [webmachinelearning/webnn#756]
Since in many cases a resource continues to exist for a period of time while processing finishes I think it will be less confusing to developers to name these methods `release()`.

**Comment by bbernhar** [comment]
I'm unsure how WebNN's "destroy()" meaning differs from WebGPU's "destroy()" to justify a new name. Web developers should be familiar with "destroy" meaning [delayed destruction](https://www.w3.org/TR/webgpu/#buffer-destruction). I could see `release()` being specifically useful for interop to mean "expire".

**Comment by inexorabletash** [comment]
Bikeshedding: we could also use `dispose()`

This would align with Symbol.dispose from https://github.com/tc39/proposal-explicit-resource-management

**Comment by reillyeon** [comment]
I have to say I don't feel particularly strongly about any of `destroy()`, `dispose()` or `release()`. Aligning with `Symbol.dispose` seems like the strongest argument for any of these words.

**Comment by anssiko** [comment]
Could we address this by updating the [`destroy()` prose](https://www.w3.org/TR/webnn/#api-mlcontext-destroy) a la [WebGPU's](https://www.w3.org/TR/webgpu/#buffer-destruction)? It feels like developers' cognitive load would be lowered if we stay aligned with the WebGPU's verbiage?

**Comment by reillyeon** [comment]
I agree that if most of the platform (particularly WebGPU) has chosen `destroy()` then we shouldn't introduce additional words even if they are technically more correct.

If there are no objections I am going to close this issue and paint the word "destroy" on the side of the bike shed.

**Comment by inexorabletash** [comment]
I don't see any objections, so I'm going to close this on @reillyeon's behalf. Please comment or re-open if more discussion is warranted.
